File: BOJ/4690.rs
fn main() {
    for a in 2..101 {
        for b in 2..101 {
            for c in b + 1..101 {
                for d in c + 1..101 {
                    let left = a * a * a;
                    let right = b * b * b + c * c * c + d * d * d;

                    if left == right {
                        println!("Cube = {a}, Triple = ({b},{c},{d})");
                        break;
                    } else if left < right {
                        break;
                    }
                }
            }
        }
    }
}

File: BOJ/4690.cpp
#include <iostream>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    for (int a = 2; a <= 100; a++) {
        for (int b = 2; b < a; b++) {
            for (int c = b; c < a; c++) {
                for (int d = c; d < a; d++) {
                    if (a * a * a == b * b * b + c * c * c + d * d * d) {
                        cout << "Cube = " << a << ", Triple = (" << b << ',' << c << ',' << d << ")\n";
                    }
                }
            }
        }
    }

    return 0;
}
